velox/common/caching/SsdFile.cpp

```diff
@@ -837,28 +805,55 @@ void SsdFile::checkpoint(bool force) {
   }
 }
 
+void SsdFile::addToCheckpoint(
+    std::unique_ptr<folly::IOBuf> buffer,
+    bool force) {
+  if (checkpointBuffer_ == nullptr) {
```

Comment:
I think we could use LocalWriteFile::append(std::string_view data) api. And you just keep a velox buffer allocated from cache pool and keep it around until checkpoint is done.
```
allocateCheckpointBuffer
freeCheckpointBuffer
```
different versions of
```
appendToCheckpointBuffer() {
   VELOX_CHECK_NOT_NULL(checkpointBuffer_);
    compute new size
    VELOX_CHECK_LE(new_size, checkpointBufferSize_);
    maybeFlushCheckpointBuffer(new size)
    copy to checkpoint buffer 
}

maybeFlushCheckpointBuffer(new size) {
   if (checkpointBufferUsedBytes_ + new size > limit) {
      flushCheckpointBuffer
      checkpointBufferUsedBytes_ = 0;
   }
}
```

Reply:
> we could use LocalWriteFile::append(std::string_view data) api. And you just keep a velox buffer allocated from cache pool and keep it around until checkpoint is done.

could you elaborate?